### PR TITLE
feat: [#187010596] Create License Reinstatement Quick Action CMS Content Collection

### PR DESCRIPTION
**Pull request by cbrunefearless**
## Description
 Create License Reinstatement Quick Action CMS Content Collection

### Ticket

[#187010596](https://www.pivotaltracker.com/story/show/)

## Code author checklist

- [X] I have performed a self-review of my code
- [X] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [X] I have not used any relative imports
- [X] I have checked for and removed instances of unused config from CMS
- [X] I have pruned any instances of unused code
- [X] I have not added any markdown to labels, titles and button text in config
- [X] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [X] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: content/src/quick-action-license-reinstatements/hvacr-license-reinstatement.md [hunk 1]
```diff
@@ -10,26 +10,28 @@ summaryDescriptionMd: "\n"
...
 
 ### Reinstatement Requirements
 
-*  `EIN|EIN` 
-* Social Security number (SSN)
-* Suspended license number
-* List of licenses or certificates (past and present) 
-* Certification of employment, listing all jobs held during suspension
-* Certificate of completion for continuing education 
-*  `Surety bond|surety-bond`
```
Comment (cbrunefearless) [inline content/src/quick-action-license-reinstatements/hvacr-license-reinstatement.md:24]: nothing changed

File: web/src/lib/static/loadQuickActionLicenseReinstatements.test.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        contentMd: "Some content description 2",
+        callToActionLink: "CallToActionLink2",
+        callToActionText: "CallToActionText2",
+        form: "Form2",
+      });
+    });
+  });
+
+  describe("loadAllQuickActionLicenseReinstatementsUrlSlugs", () => {
+    it("returns quick actions license reinstatement by matching given url slug", () => {
```
Comment (somabadri) [inline web/src/lib/static/loadQuickActionLicenseReinstatements.test.ts:130]: [pebble] the test says matching url slug but the test is returning all quick action reinstatements?

File: web/src/components/dashboard/QuickActionTile.tsx [hunk 3]
```diff
@@ -14,7 +14,12 @@ interface TaskProps {
   quickAction: QuickActionTask;
 }
 
-export const QuickActionTile = (props: LinkProps | TaskProps): ReactElement => {
+interface LicenseReinstatementProps {
+  type: "license";
```
Comment (somabadri) [inline web/src/components/dashboard/QuickActionTile.tsx:18]: [pebble] why does this need to be a prop if it's always going be of license type, do you see it changing in the future to include more types?

**Review by somabadri — APPROVED** [review]
Looks good to me, just left a few comments. Good job Chris!

File: web/src/components/dashboard/QuickActionTile.tsx (hunk 3 above)
Comment (jphechter) [reply web/src/components/dashboard/QuickActionTile.tsx:18]: @cbrunefearless @somabadri Initially, I was thinking that this wasn't necessary. But looking at the [next story we're going to tackle](https://www.pivotaltracker.com/story/show/186880077) related to the reinstatement quick action, we're going to be building in functionality based on a given license type being expired. While these are quick actions that will drive to tasks, we only want them to show up conditionally based on the related license being expired. We'll want to be able to check that a given quick action is of type `license` in order to conditionally show it.

File: web/src/lib/types/types.ts [hunk 4]
```diff
@@ -222,6 +222,15 @@ export interface QuickActionLink extends QuickAction {
...
 }
 
+export interface QuickActionLicenseReinstatement extends QuickAction {
+  urlSlug: string;
+  contentMd: string;
+  callToActionLink: string | undefined;
+  callToActionText: string | undefined;
+  form: string | undefined;
+  summaryDescriptionMd: string;
+}
```
Comment (jphechter) [inline web/src/lib/types/types.ts:232]: [sand] I'm not sure if it makes sense for there to be a `QuickActionLicenseReinstatement` type. This is really just the same thing as a `QuickActionTask`. The only difference between these components is that one will conditionally render based on the status of a license.

It looks like this is already merged, but in the future if you feel strongly about creating a new type like this I might have just had `QuickActionLicenseReinstatement` extend `QuickActionTask`.

File: web/src/lib/utils/markdownReader.ts [hunk 5]
```diff
@@ -180,6 +194,19 @@ type LicenseGrayMatter = {
...
+  urlSlug: string;
+  callToActionLink: string;
+  callToActionText: string;
+  form: string;
+  icon: string;
+  industryIds: string[];
+  sectorIds: string[];
+  applyToAllUsers: boolean;
+  summaryDescriptionMd: string;
+};
```
Comment (jphechter) [inline web/src/lib/utils/markdownReader.ts:208]: Same as the other comment, this is this same as `QuickActionTaskGrayMatter`.

File: content/src/quick-action-license-reinstatements/hvacr-license-reinstatement.md (hunk 1 above)
Comment (seidior) [reply content/src/quick-action-license-reinstatements/hvacr-license-reinstatement.md:24]: Formatting. It's a disagreement between Prettier and Decap about how to do lists.

**Pull request merged**